src/animations.ts

```diff
@@ -592,4 +597,11 @@ export class Animation {
 
     updateParticle();
   }
+
+  /**
+   * @param cancelled If `true`, stop evolution-related anims because the player cancelled evolution.
+   */
+  public setEvolutionCancelled(cancelled: boolean) {
+    this.evolutionCancelled = cancelled;
+  }
```

Comment:
NIT: Use a a `get/set` instead:

```suggestion
export class Animation {
  private scene: BattleScene;
  private _evolutionCancelled: boolean = false;

  //...

  public set evolutionCancelled(cancelled: boolean) {
    this._evolutionCancelled = cancelled;
  }
  
  public get evolutionCancelled() { 
    return this._evolutionCancelled;
  }
  //...
}
```